tests/lua/cartesi-machine-tests.lua

```diff
@@ -715,6 +715,16 @@ local function check_test_result(ctx)
     end
 end
 
+local function check_job_results(contexts)
+    for _, ctx in ipairs(contexts) do
+        if ctx.failed then
+            for _, e in pairs(ctx.errors) do
+                print(string.format("%s: %s", ctx.ram_image, e))
+            end
+        end
+    end
+end
+
 local function run_parallel(contexts)
```

Comment:
This "threading" logic could be simpler, I propose something like this:
(We can do this on a follow up Issue of course)

```
local unistd = require('posix.unistd')
local sys_wait = require('posix.sys.wait')

local jobs_table = {
	{{'/usr/bin/sleep', {'1'}}, 1},
	{{'/usr/bin/sleep', {'2'}}, 2},
	{{'/usr/bin/sleep', {'3'}}, 3},
	{{'/bin/false', {}}, 0},
}

local function spawn(job)
	local pid, err = unistd.fork()
	if pid == 0 then
		os.exit(unistd.exec(job[1], job[2]))
	end
	return pid
end

local function run_parallel(jobs, limit)
	table.sort(jobs, function(a, b) return b[2] < a[2] end) -- slower first
	local job
	local it, tb, i = ipairs(jobs)
	local running = 0
	local limit = limit or 1

	local function drain(x)
		local pid, reason, rc = sys_wait.wait(-1)
		while pid and pid ~= (x and 0) do
			print(pid, reason, rc)
			running = running - 1
			pid, reason, rc = sys_wait.wait(-1, sys_wait[x])
		end
	end

	while true do
		while running < limit do
			i, job = it(tb, i)
			if not job then goto done end
			local pid = spawn(job[1])
			print(pid)
			running = running + 1
		end
		drain('WNOHANG')
	end
	::done::
	drain()
end

run_parallel(jobs_table, 2)
```

Reply:
Removed, will be correctly fixed on another PR